### ISSUE TITLE
Unreachable error condition in NextMsg()

### ISSUE THREAD

**Issue by mcqueary** [nats-io/nats#125]
TL;DR `delivered > max` can never be `true` in this function. Let's remove that check and its associated error.

`NextMsg(timeout)` bails out with an error if the subscription has become invalid (closed). Reasons the subscription can become invalid/closed are: 

- Slow consumer
- Max messages reached
- Connection closed 

So, consider the following code:
```go
	select {
	case msg, ok = <-mch:
		if !ok {
			return nil, ErrConnectionClosed
		}
		delivered := atomic.AddUint64(&s.delivered, 1)
		if max > 0 {
			if delivered > max {
				return nil, errors.New("nats: Max messages delivered")
			}
			// Remove subscription if we have reached max.
			if delivered == max {
				nc.mu.Lock()
				nc.removeSub(s)
				nc.mu.Unlock()
			}
		}

	case <-t.C:
		return nil, ErrTimeout
	}
```
The error condition in the middle (`delivered > max`) is not possible. Only one  message may be retrieved at a time. When `delivered == max`, the subscription is closed/removed (`nc.removeSub(s)`), which means the next time NextMsg(timeout) is called for this subscription, it will bail before it ever gets to that line. Let's get rid of this unreachable check.

**Comment by derekcollison** [comment]
Multiple Goroutines could be calling into NextMsg() at the same time. And since mch is not necessarily unbuffered, this is possible IMO.

**Comment by mcqueary** [comment]
Good point. I wasn't thinking about concurrency there. Sounds like a good test :)
